### ISSUE TITLE
I want to add another level with custom header since this library depends on CollectionView

### ISSUE THREAD

**Issue by iballan** [rechsteiner/Parchment#272]
I am thinking of using this library for doing such UI

![screen shot 2018-11-05 at 11 41 19 am](https://user-images.githubusercontent.com/735154/47987246-19894700-e0f0-11e8-9cda-2e4be4016102.png)

1- is a google maps view
2- Category
3- Subcategory
4- List of products

when user clicks category in (2), i will reload Parchment with the new subcategories (3+4).
So 3-4 can be done using Parchment and it works fine
Can I add Categories (2) and Map View (1) as header for Parchment ?
I would like to get a hint from you @rechsteiner as is it doable ? and where should I start from using Parchment ?

**Comment by rechsteiner** [comment]
Hi @iballan! This should definitely be possible 🙂 I've just added another example for how to add a header view above the menu: https://github.com/rechsteiner/Parchment/blob/master/HeaderExample/ViewController.swift. Here's a video of how that looks: https://www.youtube.com/watch?v=mPwk_ky7q-Q. 

Does that work for you?

**Comment by rechsteiner** [comment]
Did you figure out how to do this?

**Comment by iballan** [comment]
Thank you so much for the example.

What i was trying to achieve from the above screenshot:
**1- is a google maps** view (exists in the example as header)
**2- Category** (not exists in the example, and i could not handle how and where should I add it)
**3- Subcategory** (exists in the example as the `PagingIndexItem(index: index, title: "View \(index)") as! T`)
**4- List of products** (exists in the example as `TableViewController()`)

What i have done till i am able to find out a solution:

![iphone XS Screenshot](https://user-images.githubusercontent.com/735154/50401498-0f0a3600-07a0-11e9-8edc-ad9745e826c0.jpg)
I used [Pulley](https://github.com/52inc/Pulley) for separating the map controller from the list controller.

I have added **2-Category** and **3-Subcategory** layers as sections for collection view. The last section is the **4-List of products**. I reload the product's section when user selects the subcategory.

I already finished the app in this manner. I will rewrite this section as I find out solution for this problem.

Thank you so much for the example. it would be great to have such feature in your library.
